File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        prefix_sum = 0
        sum_count = {0 : 1} # initial for "0" sum occured "1" time.

        for n in nums:
            prefix_sum += n

            # prefix - k exist in sum_dict?
                # same value is existing in dict
            if prefix_sum - k in sum_count:
                count += sum_count[prefix_sum - k]

            sum_count[prefix_sum] = sum_count.get(prefix_sum, 0) + 1
        
        return count

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cs
public class Solution {
    public int SubarraySum(int[] nums, int k) {
        int count = 0;
        int prefixSum = 0;

        var sumCount = new Dictionary<int, int>();
        sumCount[0] = 1;

        foreach (int n in nums)
        {
            prefixSum += n;

            if (sumCount.ContainsKey(prefixSum - k))
            {
                count += sumCount[prefixSum - k];
            }

            if (sumCount.ContainsKey(prefixSum)) sumCount[prefixSum]++;
            else sumCount[prefixSum] = 1;
        }
        return count;
    }
    
}